modules/apps/dynamic-data-mapping/dynamic-data-mapping-expression/src/main/java/com/liferay/dynamic/data/mapping/expression/internal/DDMExpressionEvaluatorVisitor.java

```diff
@@ -134,11 +134,20 @@ public Object visitEqualsExpression(
 		Object object1 = visitChild(context, 0);
 		Object object2 = visitChild(context, 2);
 
-		if ((object1 instanceof Number) && (object2 instanceof Number)) {
-			BigDecimal bigDecimal1 = new BigDecimal(object1.toString());
-			BigDecimal bigDecimal2 = new BigDecimal(object2.toString());
+		if ((object1 instanceof Number) || (object2 instanceof Number)) {
```

Comment:
If they are both string, this code won't parse for BigDecimal. Is this the intention?

Reply:
@gabrielwas  Yes. In that case, both data will be compared like any other _Object_ in: `Objects.equals(object1, object2);`